rfc/text/advanced/ap_10_rpc_progressive_calls.md

```diff
@@ -1,5 +1,249 @@
+
+**Feature Announcement**
+
+Support for this advanced feature MUST be announced by *Callers* (`role := "caller"`), *Callees* (`role := "callee"`) 
+and *Dealers* (`role := "dealer"`) via
+
+{align="left"}
+HELLO.Details.roles.<role>.features.progressive_calls|bool := true
+
+Progressive Calls can work only if all three nodes support and announced this feature. 
+
+Cases when *Caller* sends `CALL` message with `progressive_calls := true` without announcing it during `HELLO` 
+handshake MUST be treated as *PROTOCOL ERRORS* and underlying WAMP connections must be aborted with 
+`wamp.error.protocol_violation` error reason.
+
+Cases when *Caller* sends `CALL` message with `progressive_calls := true` to *Dealer*, that did not announce 
+progressive calls support during `WELCOME` handshake MUST be treated as *PROTOCOL ERRORS* and underlying WAMP 
+connections must be aborted with `wamp.error.protocol_violation` error reason.
+
+Cases when *Caller* sends `CALL` message with `progressive_calls := true` to the *Dealer* that supports this feature,
+which then must be routed to the *Callee* which doesn't support progressive calls MUST be treated as *APPLICATION ERRORS*
+and *Dealer* MUST respond to *Caller* with `wamp.error.feature_not_supported` error message.
+
+**Message Flow**
+
+The message flow for progressive call when *Callee* waits for all chunks before processing and sending single result:
+
+{align="left"}
+,------.            ,------.                 ,------.
+|Caller|            |Dealer|                 |Callee|
+`--+---'            `--+---'                 `--+---'
+   |  CALL (progress)  |                        |
+   | ----------------->|                        |
+   |                   | INVOCATION (progress)  |
+   |                   | ---------------------->|
+   |  CALL (progress)  |                        |
+   | ----------------->|                        |
+   |                   | INVOCATION (progress)  |
+   |                   | ---------------------->|
+   |                   |                        |
+   |       ...         |       ...              |
+   |                   |                        |
+   |  CALL (final)     |                        |
+   | ----------------->|                        |
+   |                   | INVOCATION (final)     |
+   |                   | ---------------------->|
+   |                   |                        |
+   |                   |  YIELD or ERROR        |
+   |                   | <----------------------|
+   |                   |                        |
+   |  RESULT or ERROR  |                        |
+   | <-----------------|                        |
+,--+---.            ,--+---.                 ,--+---.
+|Caller|            |Dealer|                 |Callee|
+`------'            `------'                 `------'
+
+As `progressive` call is still the same call *Callee* must send the same `Request|id` with every `CALL` and *Dealer*
```

Comment:
May I suggest: "All `CALL`, `INVOCATION`, `YIELD`, `RESULT` and `ERROR` messages belonging to the same overall progressive call MUST use the same `Request|Id`".

Reply:
Hm... Is it not confusing? I mean `Request|Id` is different between CALL — INVOCATION — YIELD — RESULT. So it should be the same but within a related message type.